modules/govuk_mysql/manifests/xtrabackup/backup.pp

```diff
@@ -0,0 +1,113 @@
+# == Class: Govuk_mysql::Xtrabackup::Backup
+#
+# Creates base and incremental backups which are streamed to an S3 bucket.
+# Based upon this MariaDB blog post: https://mariadb.com/blog/streaming-mariadb-backups-cloud
+#
+# === Parameters
+#
+# [*aws_access_key_id*]
+#   The AWS access key used to access the S3 bucket.
+#
+# [*aws_secret_access_key*]
+#   The AWS secret access key used to access the S3 bucket.
+#
+# [*aws_region*]
+#   The region where the S3 bucket lives.
+#   Default: eu-west-1
+#
+# [*s3_bucket_name*]
+#   The unique name of the S3 bucket.
+#
+# [*encryption_key*]
+#   The encryption key used to encrypt the backups.
+#
+# [*base_backup_cron_minute*]
+#   The minute(s) that the daily base backup(s) run(s).
+#   Default: 10
```

Comment:
I think we could remove the defaults from the documentation since they're likely to get out of date quickly.